test/realm_test.dart

```diff
@@ -1647,3 +1631,13 @@ When newWhen([tz.TZDateTime? time]) {
 extension _IterableEx<T> on Iterable<T> {
   Iterable<T> except(T exclude) => where((o) => o != exclude);
 }
+
+void _expectAllWritesToThrow<T>(Realm realm, String exceptionMessage) {
```

Comment:
```suggestion
void _expectAllWritesToThrow<T extends Exception>(Realm realm, String exceptionMessage) {
```

Whe should be throwing a RealmException on write. I had suggested previously to check that in all writes and throw it instead of RealmClosedError.

Reply:
We decided in a call that no additional check for r`ealm.isClosed` is needed in write methods. We already check if the handle is released.